CONTRIBUTING.md

```diff
@@ -39,11 +40,25 @@ stack test
 stack bench
 ```
 
+
 ## Troubleshooting
 
-- If you're using Stack, make sure you have version >= 1.6.1 ([1], [2])
+- If you're using Stack, make sure you have version >= 1.6.1
+  ([stack#3524](https://github.com/commercialhaskell/stack/issues/3624),
+  [stack#3345](https://github.com/commercialhaskell/stack/issues/3345)).
+
+
+## Sending Pull Requests
+
+When you send a pull request, please:
 
+- Link to libraries@haskell.org discussion thread if you are changing the public
+  API.
 
-[1] https://github.com/commercialhaskell/stack/issues/3624
+- Run the benchmarks and include the results in the commit message and Pull
+  Request.
```

Comment:
Benchmarks aren't entirely mandatory in all cases. I think we could soften this some.

> If you are requesting a change that is likely to affect performance, we will be able to evaluate it better if you include the results of running the benchmarks before and after. If the current benchmarks cannot demonstrate a desired difference, please try to add one or more new benchmarks to do so. If there are significant changes, please include the benchmark results in your commit message.

Reply:
Done.